l10n_it_central_journal/wizard/print_giornale.py

```diff
@@ -0,0 +1,165 @@
+         ('printed', 'Printed')],
+        'State',
+        default='print',
+        readonly=True)
+    journal_ids = fields.Many2many(
+        'account.journal',
+        'giornale_journals_rel',
+        'journal_id',
+        'giornale_id',
+        default=_get_journal,
+        string='Journals',
+        required=True)
+    target_move = fields.Selection([('all', 'All'),
+                                    ('posted', 'Posted'),
+                                    ('draft', 'Draft')],
+                                   'Target Move', default='all')
+    fiscal_page_base = fields.Integer('Last printed page', required=True)
+    start_row = fields.Integer('Start row', required=True)
+    year_footer = fields.Char(string='Year for Footer',
+                              help="Value printed near number "
+                                   "of page in the footer")
+
+    @api.onchange('date_move_line_from_view')
+    def get_year_footer(self):
+        if self.date_move_line_from_view:
+            self.year_footer = datetime.strptime(self.date_move_line_from_view,
+                                                 "%Y-%m-%d").year
+
+    @api.onchange('daterange')
+    def on_change_daterange(self):
+        if self.daterange:
+            date_start = datetime.strptime(
+                self.daterange.date_start, "%Y-%m-%d").date()
+            date_end = datetime.strptime(
+                self.daterange.date_end, "%Y-%m-%d").date()
+            if self.daterange.date_last_print:
+                date_last_print = datetime.strptime(
+                    self.daterange.date_last_print, "%Y-%m-%d").date()
+                self.last_def_date_print = date_last_print
+                date_start = (date_last_print + timedelta(days=1)).__str__()
+            else:
+                self.last_def_date_print = None
+            self.date_move_line_from = date_start
+            self.date_move_line_from_view = date_start
+            self.date_move_line_to = date_end
+            if self.daterange.progressive_line_number != 0:
+                self.start_row = self.daterange.progressive_line_number + 1
+            else:
+                self.start_row = self.daterange.progressive_line_number
+            self.progressive_debit2 = self.daterange.progressive_debit
+            self.progressive_credit = self.daterange.progressive_credit
+
+            if self.last_def_date_print == self.daterange.date_end:
+                self.date_move_line_from_view = self.last_def_date_print
+
+    def print_giornale(self):
+        wizard = self
+        if wizard.target_move == 'all':
+            target_type = ['posted', 'draft']
+        else:
+            target_type = [wizard.target_move]
+        move_line_obj = self.env['account.move.line']
+        move_line_ids = move_line_obj.search([
+            ('date', '>=', wizard.date_move_line_from),
+            ('date', '<=', wizard.date_move_line_to),
+            ('move_id.state', 'in', target_type)
```

Comment:
almeno in quella NON definitiva aggiungerei il filtro sui journal. si possono togliere dall'interfaccia, ma li stampa sempre tutti
('journal_id', 'in', [j.id for j in wizard.journal_ids])

@eLBati cosa ne pensi? In quella definitiva l'utente deve  stampare tutti i sezionali che ha non flaggato con "escludi da libro giornale", ma in quella di prova può avere senso lasciare all'utente la scelta, magari esplicitandolo.

Reply:
Se nel wizard è possibile selezionare i registri, è fuorviante poi non usarli, in ogni caso. Quindi sì, in quella non definitiva dovrebbero essere considerati. Se in quella definitiva non vengono in ogni caso considerati, allora non devono comparire nel wizard.

Reply:
Il form è lo stesso, ma ci sono due bottoni differenti. Abbiamo aggiunto un commento che avvisa l'utente che in caso di stampa definitiva il filtro sui registri verrà ignorato.